server/container_create_linux.go

```diff
@@ -322,6 +324,11 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrIface.Contai
 		return nil, err
 	}
 
+	cdiDevices, err := getRequestedCDIDevices(ctr.Config().GetAnnotations())
```

Comment:
can we move these pieces into `pkg/container/devices.go`?

Reply:
I moved the code over there, triggering CDI injection from `container.SpecAddDevices()`.

Reply:
I'll take a look at those new integration et. al test failures in the most recent CI run...